src/Core/Tokens/Token.cs

```diff
@@ -26,11 +27,28 @@ public Token RemovePrefix(string expectedPrefix)
         return new Token(_token[expectedPrefix.Length..]);
     }
 
-    public Token ProtectWith(IDataProtector dataProtector) =>
-        new(dataProtector.Protect(ToString()));
 
-    public Token UnprotectWith(IDataProtector dataProtector) =>
-        new(dataProtector.Unprotect(ToString()));
+    public Token ProtectWith<T>(IDataProtector dataProtector, ILogger<T> logger)
```

Comment:
Is the only reason that we have to make this a generic because of `ILogger<T>`?

Reply:
Yes

Reply:
You don't have to make this generic for that. If you pass in the `ILogger<T>` but this parameter is `ILogger` it will still keep the category name associated with it.